File: leetcode/1291-sequential-digits/1291-sequential-digits.js
var sequentialDigits = function(low, high) {
    const res = [];

    const lowDigit = String(low).length;
    const highDigit = String(high).length;

    for (let digits = lowDigit; digits <= highDigit; digits++) {
        for (let start = 1; start < 10; start++) {
            if (start + digits > 10) {
                break;
            }

            let num = start;
            let prev = start;

            for (let i = 1; i < digits; i++) {
                num = num * 10 + (++prev);
            }

            if (num >= low && num <= high) {
                res.push(num);
            }
        }
    }

    return res;
};

File: leetcode/1291-sequential-digits/1291-sequential-digits.java
public class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        List<Integer> res = new ArrayList<>();
        int lowDigit = String.valueOf(low).length();
        int highDigit = String.valueOf(high).length();

        for (int digits = lowDigit; digits <= highDigit; digits++) {
            for (int start = 1; start < 10; start++) {
                if (start + digits > 10) {
                    break;
                }
                int num = start;
                int prev = start;
                for (int i = 1; i < digits; i++) {
                    num = num * 10 + (++prev);
                }
                if (num >= low && num <= high) {
                    res.add(num);
                }
            }
        }
        return res;
    }
}